main/externalData/balances/worker.ts

```diff
@@ -51,12 +52,21 @@ function sendToMainProcess (data: any) {
   }
 }
 
-async function tokenBalanceScan (address: Address, tokensToOmit: Token[] = [], chains?: number[]) {
+async function updateBlacklist (address: Address, chains: number[]) {
+  try {
+    const blacklistTokens = tokenLoader.getBlacklist(chains)
+    sendToMainProcess({ type: 'tokenBlacklist', address, tokens: blacklistTokens })
+  } catch (e) {
+    log.error('error updating token blacklist', e)
+  }
+}
+
+async function tokenBalanceScan (address: Address, tokensToOmit: Token[] = [], chains: number[]) {
   try {
     // for chains that support multicall, we can attempt to load every token that we know about,
     // for all other chains we need to call each contract individually so don't scan every contract
     const eligibleChains = (chains || await getChains()).filter(chainSupportsScan)
-    const tokenLists = eligibleChains.map(chainId => tokenLoader.getTokens(chainId))
+    const tokenLists = eligibleChains.map(chainId => tokenLoader.getTokens([chainId]))
```

Comment:
```suggestion
    const tokenLists = tokenLoader.getTokens(eligibleChains)
```

Reply:
Didn't realise it literally reduces the 2d array below... 🤦